packages/react-inbox/src/reducer.ts

```diff
@@ -64,8 +64,27 @@ export default (state: InboxState = {}, action) => {
       startCursor: action?.payload?.startCursor,
       messages: action?.payload?.appendMessages ? [
         ...(state.messages ??[]),
-        ...newMessages
-      ] : newMessages
+        ...newMessages,
+      ] : newMessages,
+    };
+  }
+
+  case "inbox/MARK_MESSAGE_UNREAD": {
+    const unreadMessageCount = (state.unreadMessageCount ?? 1) + 1;
```

Comment:
could we make this sort of decision on the dispatch end? Like could this be part of action payload rather than reducer logic?

Reply:
this should be `const unreadMessageCount = (state.unreadMessageCount ?? 0) + 1;`

the other way was to prevent it going to `-1`

Reply:
but i liked your idea on Max/Min

Reply:
Nice catch. Thank you. Fixed